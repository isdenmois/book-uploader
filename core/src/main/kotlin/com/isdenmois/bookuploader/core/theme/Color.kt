package com.isdenmois.bookuploader.core.theme

import androidx.compose.material.Colors
import androidx.compose.material.MaterialTheme
import androidx.compose.ui.graphics.Color

val DarkColorPalette = AppColors(
    isLight = false,

    background = Color(0xff1f1d2b),
    searchBackground = Color(0xff525298),
    uploadBackground = Color(0xff805da7),
    profileBackground = Color(0xffd95333),

    tabsBackground = Color(0xff212230),
    secondaryBackground = Color(0xffE0E0E0),
    progressBackground = Color(0xffDEE2ED),
    modalBackground = Color(0xff212230),

    searchText = Color(0xfffefefe),
    uploadText = Color(0xfffefefe),
    profileText = Color(0xfffefefe),
    tabText = Color(0xff7b7c89),

    searchSelected = Color(0xff525298),
    uploadSelected = Color(0xff9c4ec2),
    profileSelected = Color(0xffff7128),

    primary = Color(0xff9c4ec2),
    invertedText = Color.White,
    secondary = Color(0xff92949b),
    search = Color(0xff8C93A3),
    input = Color(0xff252836),
    deleteButton = Color(0xff9BA8CA),
    success = Color(0xff22b07d),
    error = Color(0xffd8125d),
)

val LightColorPalette = AppColors(
    isLight = true,

    background = Color(0xffF9FAFE),
    searchBackground = Color(0xffB8D5F9),
    uploadBackground = Color(0xffC2B9F9),
    profileBackground = Color(0xffFFAD99),
    tabsBackground = Color.White,
    secondaryBackground = Color(0xffE0E0E0),
    progressBackground = Color(0xffDEE2ED),
    modalBackground = Color.White,

    searchText = Color(0xff092D5D),
    uploadText = Color(0xff15085E),
    profileText = Color(0xff661400),
    tabText = Color(0xff121417),

    searchSelected = Color(0xff5396EE),
    uploadSelected = Color(0xff6851F0),
    profileSelected = Color(0xffFF6842),

    primary = Color(0xff15085E),
    invertedText = Color.White,
    secondary = Color(0xff787D87),
    search = Color(0xff8C93A3),
    input = Color.White,
    deleteButton = Color(0xff9BA8CA),
    success = Color(0xff48B755),
    error = Color(0xffFD4463),
)

/**
 * A Material [Colors] implementation which sets all colors to [debugColor] to discourage usage of
 * [MaterialTheme.colors] in preference to [AppTheme.colors].
 */
fun debugColors(
    darkTheme: Boolean,
    debugColor: Color = Color.Magenta
) = Colors(
    primary = debugColor,
    primaryVariant = debugColor,
    secondary = debugColor,
    secondaryVariant = debugColor,
    background = debugColor,
    surface = debugColor,
    error = debugColor,
    onPrimary = debugColor,
    onSecondary = debugColor,
    onBackground = debugColor,
    onSurface = debugColor,
    onError = debugColor,
    isLight = !darkTheme
)
