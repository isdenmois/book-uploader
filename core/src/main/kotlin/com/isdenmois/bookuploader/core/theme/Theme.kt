package com.isdenmois.bookuploader.core.theme

import androidx.compose.foundation.isSystemInDarkTheme
import androidx.compose.material.MaterialTheme
import androidx.compose.runtime.Composable
import androidx.compose.runtime.CompositionLocalProvider
import androidx.compose.runtime.remember
import androidx.compose.runtime.staticCompositionLocalOf

object AppTheme {
    val colors: AppColors
        @Composable
        get() = LocalAppColors.current
}

@Composable
fun AppTheme(
    darkTheme: Boolean = isSystemInDarkTheme(),
    content: @Composable () -> Unit
) {
    val colors = if (darkTheme) DarkColorPalette else LightColorPalette

    ProvideAppColors(colors) {
        MaterialTheme(
            colors = debugColors(darkTheme),
            typography = Typography,
            shapes = Shapes,
            content = content
        )
    }
}

@Composable
fun ProvideAppColors(
    colors: AppColors,
    content: @Composable () -> Unit
) {
    val colorPalette = remember {
        // Explicitly creating a new object here so we don't mutate the initial [colors]
        // provided, and overwrite the values set in it.
        colors.copy()
    }
    colorPalette.update(colors)
    CompositionLocalProvider(LocalAppColors provides colorPalette, content = content)
}

private val LocalAppColors = staticCompositionLocalOf<AppColors> {
    error("No AppColorPalette provided")
}
