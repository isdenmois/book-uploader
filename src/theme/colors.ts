export const primary = '#000';
export const secondary = '#eee';
export const invert = '#fff';

export const green = '#8BC34A';
export const red = '#f44336';
export const darkBlue = '#007ac1';
export const blue = '#2196f3';
export const lightBlue = '#67daff';

export const Background = '#F9FAFE';
export const SearchBackground = '#B8D5F9';
export const UploadBackground = '#C2B9F9';
export const ProfileBackground = '#FFAD99';
export const TabsBackground = 'white';

export const SearchText = '#092D5D';
export const UploadText = '#15085E';
export const ProfileText = '#661400';
export const TabText = '#121417';

export const SearchSelected = '#5396EE';
export const UploadSelected = '#6851F0';
export const ProfileSelected = '#FF6842';

export const Secondary = '#787D87';
export const Search = '#8C93A3';
export const Input = 'white';
export const DeleteButton = '#9BA8CA';
export const Success = '#48B755';
export const Error = '#FD4463';
