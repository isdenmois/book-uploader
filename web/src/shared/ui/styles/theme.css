button,
input {
  all: unset;
}

button:focus {
  outline: none;
}

body {
  background-color: var(--background);
}

#app {
  max-width: 600px;
  margin: 0 auto;
  height: 100vh;
  overflow: hidden;
  display: flex;
  flex-direction: column;
}

h1 {
  font-size: 24px;
  color: var(--profileText);
}

.book-list > div:not(:first-child) {
  @apply mt-4;
}

.input {
  background: #ffffff;
  box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.25);
  border-radius: 10px;
  padding-left: 4px;
}
