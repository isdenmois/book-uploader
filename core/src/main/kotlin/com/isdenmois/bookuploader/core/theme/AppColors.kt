package com.isdenmois.bookuploader.core.theme

import androidx.compose.runtime.Stable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.setValue
import androidx.compose.ui.graphics.Color

@Stable
class AppColors(
    background: Color,
    searchBackground: Color,
    uploadBackground: Color,
    profileBackground: Color,
    tabsBackground: Color,
    secondaryBackground: Color,
    progressBackground: Color,
    modalBackground: Color,

    searchText: Color,
    uploadText: Color,
    profileText: Color,
    tabText: Color,

    searchSelected: Color,
    uploadSelected: Color,
    profileSelected: Color,

    primary: Color,
    invertedText: Color,
    secondary: Color,
    search: Color,
    input: Color,
    deleteButton: Color,
    success: Color,
    error: Color,
    isLight: Boolean
) {
    var background by mutableStateOf(background)
        private set
    var searchBackground by mutableStateOf(searchBackground)
        private set
    var uploadBackground by mutableStateOf(uploadBackground)
        private set
    var profileBackground by mutableStateOf(profileBackground)
        private set
    var tabsBackground by mutableStateOf(tabsBackground)
        private set
    var secondaryBackground by mutableStateOf(secondaryBackground)
        private set
    var progressBackground by mutableStateOf(progressBackground)
        private set
    var modalBackground by mutableStateOf(modalBackground)
        private set

    var searchText by mutableStateOf(searchText)
        private set
    var uploadText by mutableStateOf(uploadText)
        private set
    var profileText by mutableStateOf(profileText)
        private set
    var tabText by mutableStateOf(tabText)
        private set

    var searchSelected by mutableStateOf(searchSelected)
        private set
    var uploadSelected by mutableStateOf(uploadSelected)
        private set
    var profileSelected by mutableStateOf(profileSelected)
        private set

    var primary by mutableStateOf(primary)
        private set
    var invertedText by mutableStateOf(invertedText)
        private set
    var secondary by mutableStateOf(secondary)
        private set
    var search by mutableStateOf(search)
        private set
    var input by mutableStateOf(input)
        private set
    var deleteButton by mutableStateOf(deleteButton)
        private set
    var success by mutableStateOf(success)
        private set
    var error by mutableStateOf(error)
        private set
    var isLight by mutableStateOf(isLight)
        private set

    fun update(other: AppColors) {
        background = other.background
        searchBackground = other.searchBackground
        uploadBackground = other.uploadBackground
        profileBackground = other.profileBackground
        tabsBackground = other.tabsBackground
        secondaryBackground = other.secondaryBackground
        progressBackground = other.progressBackground
        modalBackground = other.modalBackground

        searchText = other.searchText
        uploadText = other.uploadText
        profileText = other.profileText
        tabText = other.tabText

        searchSelected = other.searchSelected
        uploadSelected = other.uploadSelected
        profileSelected = other.profileSelected

        primary = other.primary
        invertedText = other.invertedText
        secondary = other.secondary
        search = other.search
        input = other.input
        deleteButton = other.deleteButton
        success = other.success
        error = other.error
        isLight = other.isLight
    }

    fun copy(): AppColors = AppColors(
        background = background,
        searchBackground = searchBackground,
        uploadBackground = uploadBackground,
        profileBackground = profileBackground,
        tabsBackground = tabsBackground,
        secondaryBackground = secondaryBackground,
        progressBackground = progressBackground,
        modalBackground = modalBackground,

        searchText = searchText,
        uploadText = uploadText,
        profileText = profileText,
        tabText = tabText,

        searchSelected = searchSelected,
        uploadSelected = uploadSelected,
        profileSelected = profileSelected,

        primary = primary,
        invertedText = invertedText,
        secondary = secondary,
        search = search,
        input = input,
        deleteButton = deleteButton,
        success = success,
        error = error,
        isLight = isLight,
    )
}
