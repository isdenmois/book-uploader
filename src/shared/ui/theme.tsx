import React, { FC } from 'react'
import { TouchableOpacity, TouchableOpacityProps, useColorScheme } from 'react-native'
import {
  ThemeProvider as ReThemeProvider,
  createBox,
  createText,
  createTheme,
  useTheme as useThemeRN,
} from '@shopify/restyle'

import * as Colors from './colors'

export const theme = createTheme({
  spacing: {
    0.25: 2,
    0.5: 4,
    1: 8,
    2: 16,
    3: 24,
    4: 32,
    5: 40,
    6: 48,
  },
  colors: Colors.light,
  breakpoints: {},
  textVariants: {
    defaults: {
      fontSize: 16,
    },
    chip: {
      fontSize: 14,
      lineHeight: 18,
    },
    button_primary: {
      fontSize: 16,
      lineHeight: 22,
      color: 'invertedText',
    },
  },
  buttonVariants: {
    primary: {
      backgroundColor: 'profileSelected',
    },
  },
})

export type Theme = typeof theme

const darkTheme: Theme = {
  ...theme,
  colors: {
    ...theme.colors,
    ...Colors.dark,
  },
}

export const Box = createBox<Theme>()
export const Text = createText<Theme>()
export const TouchableBox = createBox<Theme, TouchableOpacityProps & { children: any }>(TouchableOpacity)

export const ThemeProvider: FC = ({ children }) => {
  const currentMode = useColorScheme()

  return <ReThemeProvider theme={currentMode === 'dark' ? darkTheme : theme}>{children}</ReThemeProvider>
}

export const useTheme = () => useThemeRN<Theme>()
