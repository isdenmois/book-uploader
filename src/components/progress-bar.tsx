import React from 'react';
import { Text, View } from 'react-native';
import { DynamicStyleSheet, useDynamicStyleSheet } from 'react-native-dynamic';
import { dynamicColor } from 'theme/colors';

type Props = {
  color: string;
  progress?: number;
  showAlways?: boolean;
  text?: string;
};

export function ProgressBar({ color, progress, showAlways, text }: Props) {
  const s = useDynamicStyleSheet(ds);
  if (!showAlways && (!progress || progress <= 0)) return <View style={s.empty} />;
  const progressStyle = progress < 100 ? { width: `${progress}%`, backgroundColor: color } : s.done;

  return (
    <View style={s.container}>
      <View style={s.background}>
        <View style={[s.progress, progressStyle]} />
      </View>

      <Text style={s.text}>{text || Math.round(progress) + '%'}</Text>
    </View>
  );
}

const ds = new DynamicStyleSheet({
  empty: { height: 12 },
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    height: 10,
    marginTop: 2,
  },
  background: {
    height: 5,
    position: 'relative',
    backgroundColor: dynamicColor.progressBackground,
    borderRadius: 2,
    overflow: 'hidden',
    flex: 1,
    marginTop: 1,
  },
  progress: {
    position: 'absolute',
    left: 0,
    top: 0,
    bottom: 0,
  },
  done: {
    width: '100%',
    backgroundColor: dynamicColor.success,
  },
  text: {
    color: dynamicColor.secondary,
    fontSize: 12,
    minWidth: 34,
    paddingLeft: 5,
    textAlign: 'right',
  },
});
