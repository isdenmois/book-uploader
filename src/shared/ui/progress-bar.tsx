import React from 'react'
import { Box, Theme, Text } from './theme'

type Props = {
  color: keyof Theme['colors']
  progress?: number
  showAlways?: boolean
  text?: string
}

export function ProgressBar({ color, progress, showAlways, text }: Props) {
  if (!showAlways && (!progress || progress <= 0)) return <Box height={12} />
  const width = progress < 100 ? `${progress}%` : '100%'
  const isDone = progress >= 100

  return (
    <Box flexDirection='row' alignItems='center' height={12} marginTop={0.25}>
      <Box
        height={5}
        position='relative'
        backgroundColor='progressBackground'
        borderRadius={2}
        overflow='hidden'
        flex={1}
      >
        <Box
          position='absolute'
          left={0}
          top={0}
          bottom={0}
          width={width}
          backgroundColor={isDone ? 'success' : color}
        />
      </Box>

      <Text color='secondary' fontSize={12} lineHeight={12} pl={0.5} textAlign='right'>
        {text || Math.round(progress) + '%'}
      </Text>
    </Box>
  )
}
