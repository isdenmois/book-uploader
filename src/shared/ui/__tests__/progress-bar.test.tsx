import React from 'react'
import { act, render } from '@testing-library/react-native'
import { ThemeProvider } from '../theme'
import { ProgressBar } from '../progress-bar'

test('ProgressBar', async () => {
  const { toJSON, rerender, unmount } = render(<ProgressBar color='primary' />, { wrapper: ThemeProvider })

  expect(toJSON()).toMatchSnapshot()

  rerender(<ProgressBar color='primary' showAlways />)
  expect(toJSON()).toMatchSnapshot()

  rerender(<ProgressBar color='primary' progress={50} />)
  expect(toJSON()).toMatchSnapshot()

  rerender(<ProgressBar color='primary' progress={100} />)
  expect(toJSON()).toMatchSnapshot()

  await act(async () => {
    unmount()
  })
})
