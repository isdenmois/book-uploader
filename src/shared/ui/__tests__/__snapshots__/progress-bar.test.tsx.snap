// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "height": 12,
      },
    ]
  }
/>
`;

exports[`ProgressBar 2`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "height": 10,
        "marginTop": 2,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#DEE2ED",
          "borderRadius": 2,
          "display": "flex",
          "flex": 1,
          "height": 5,
          "overflow": "hidden",
          "position": "relative",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#15085E",
            "bottom": 0,
            "display": "flex",
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          },
        ]
      }
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#787D87",
          "display": "flex",
          "fontSize": 12,
          "paddingLeft": 4,
          "textAlign": "right",
        },
      ]
    }
  >
    NaN%
  </Text>
</View>
`;

exports[`ProgressBar 3`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "height": 10,
        "marginTop": 2,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#DEE2ED",
          "borderRadius": 2,
          "display": "flex",
          "flex": 1,
          "height": 5,
          "overflow": "hidden",
          "position": "relative",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#15085E",
            "bottom": 0,
            "display": "flex",
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": "50%",
          },
        ]
      }
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#787D87",
          "display": "flex",
          "fontSize": 12,
          "paddingLeft": 4,
          "textAlign": "right",
        },
      ]
    }
  >
    50%
  </Text>
</View>
`;

exports[`ProgressBar 4`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "height": 10,
        "marginTop": 2,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#DEE2ED",
          "borderRadius": 2,
          "display": "flex",
          "flex": 1,
          "height": 5,
          "overflow": "hidden",
          "position": "relative",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#48B755",
            "bottom": 0,
            "display": "flex",
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          },
        ]
      }
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#787D87",
          "display": "flex",
          "fontSize": 12,
          "paddingLeft": 4,
          "textAlign": "right",
        },
      ]
    }
  >
    100%
  </Text>
</View>
`;
